### PR TITLE
Add canarytools banner

### PR DESCRIPTION
**Pull request by vittoriaThinkst**
## Proposed changes

Add Canarytools banner component
Add variation to button to support links
Add variation to button to support icons

Screenshot 2024-04-12 at 15 43 01
Screenshot 2024-04-12 at 15 43 08

### CODE REVIEW

File: frontend_vue/src/components/base/BaseButton.vue [hunk 1]
```diff
@@ -1,11 +1,22 @@
 <template>
-  <button
+  <component
+    :is="link ? 'a' : 'button'"
```
Comment (mclmax) [inline frontend_vue/src/components/base/BaseButton.vue:3]: this seems a little weird no? what's the benefit of overloading the button to be a link / button? I think we're drifting a little too far into modularisation land here

Comment (mclmax) [reply frontend_vue/src/components/base/BaseButton.vue:3]: like... if we're just using it in one place why not just have a standalone link with some base button styling?

File: frontend_vue/src/components/ui/BannerCanarytools.vue [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+const xlContainerClasses =
+  'grid @sm:justify-center @sm:items-center @sm:grid-cols-[2fr_3fr_1fr] @sm:grid-rows-[2fr] @sm:aspect-[4/1] @md:h-[180px] @sm:bg-[position:-200px_-100px,0] @sm:bg-[size:550px,cover]';
+const mobileContainerClasses =
+  'grid-rows-[3fr_1fr] px-16 h-[280px] aspect-[1/2] bg-[position:-250px_0px,0] bg-[size:550px,cover]';
```
Comment (mclmax) [inline frontend_vue/src/components/ui/BannerCanarytools.vue:32]: we need to be careful doing this kind of thing... folks are going to reuse this elsewhere and we're going to drift back into spaghetti CSS... are we doing this kind of thing elsewhere?

**Review by mclmax — CHANGES_REQUESTED** [review]

File: frontend_vue/src/components/ui/BannerCanarytools.vue (hunk 2 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/ui/BannerCanarytools.vue:32]: I'm not sure I understood the issue.
This banner can be placed everywhere and keeps the layout as it's shown in the preview of the diff. Ideally, nobody will touch this CSS
( I'm using container queries, not media queries )

File: frontend_vue/src/components/base/BaseButton.vue (hunk 1 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/base/BaseButton.vue:3]: The idea is that we gonna have some places where the links are styled like buttons. 

I don't see the advantage of copy/pasting the BaseButton code into BaseLinkButton.
It can eventually lead to design inconsistencies, or having to fix both components every time make a change in one of them.
I think it's easier to have one line of code that takes care of this instead.

If you think it's not clear and ppl might not find it easily, I can also create a BaseLink which is simply a BaseButton component with the link prop

File: frontend_vue/src/components/ui/BannerCanarytools.vue (hunk 2 above)
Comment (mclmax) [reply frontend_vue/src/components/ui/BannerCanarytools.vue:32]: i don't want us to get into the habit of describing the CSS in variables that can be used in more than one place - else we may as well use standard CSS

File: frontend_vue/src/components/base/BaseButton.vue (hunk 1 above)
Comment (mclmax) [reply frontend_vue/src/components/base/BaseButton.vue:3]: So this one "Find out" doesn't have any styling right? it's just text? Why not just use an anchor element?

File: frontend_vue/src/components/ui/BannerCanarytools.vue (hunk 2 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/ui/BannerCanarytools.vue:32]: Oh I see, uhm, it just gets very hard to follow with tailwind ( It would be like 3 lines of classes )
but I can move them up into the template to avoid the possibility

File: frontend_vue/src/components/base/BaseButton.vue (hunk 1 above)
Comment (vittoriaThinkst) [reply frontend_vue/src/components/base/BaseButton.vue:3]: it's text with an icon, and it takes the same style as the text button.
So far this type of link is used only here. 

If you think it makes it all complicated to follow, I might just add an anchor and style it here. If we'll ever need more of these button-styled-link in the future, we'll think about another component then

Comment (mclmax) [reply frontend_vue/src/components/base/BaseButton.vue:3]: yeah i think we might be overcomplicating the button here where an simple Anchor works?

Comment (mclmax) [reply frontend_vue/src/components/base/BaseButton.vue:3]: Even an anchor component seems like a lot? like... an anchor class maybe?

**Review by mclmax — APPROVED** [review]
